### ISSUE TITLE
RRateLimiter分布式限流BUG: 不会删除ZSET数据结构的key。

### ISSUE THREAD

**Issue by victoyzz** [redisson/redisson#5098]
**Expected behavior**

无论是服务重启，还是重新加载限流配置（rateLimiter.setRate），都不会删除下列两种ZSET结构的key。
我知道这个key是用于限流统计的。permits用于存放请求记录包括时间戳，value用于统计对应剩余的令牌桶。

KEY:
   {xxx}:permits
   {xxx}:value

这样会导致一个问题，我如果加载了新的限流配置，比如一秒生成200个令牌（原先是100）；
而上述两种key不会删除或者更新，这样当新的请求进来时，还是从前面的 key:  {xxx}:value 中拿剩余的令牌；
这样就导致，明明新设置了200限流，但是还是被限流100。

**Actual behavior**

**Steps to reproduce or test case**

**Redis version**
    <dependency>
            <groupId>org.redisson</groupId>
            <artifactId>redisson-spring-boot-starter</artifactId>
            <version>3.22.0</version>
        </dependency>
**Redisson version**

**Redisson configuration**

**Comment by mrniko** [comment]
Can you share the code to reproduce it?

**Comment by victoyzz** [comment]
> 
ok.
1、When the project starts, I will load current limiting configurations for each interface.
      `rateLimiter.setRate(2, 100, 1, RateIntervalUnit.SECONDS);`

 2、In the slice, for each interface, attempt to obtain tokens from the token bucket.
      final String ofRateLimiter = RedisKeyConstants + key;
        RRateLimiter rateLimiter = redisson.getRateLimiter(ofRateLimiter);
        if (!rateLimiter.tryAcquire()) {
            return Response.createError("接口访问超出频率限制，请稍后重试");
        }
 
 3、after, I modified the current limiting configuration of the interface and then reloaded and executed     `rateLimiter.setRate(2, 200, 1, RateIntervalUnit.SECONDS);`
 
4、Then, in the slice, when each interface requests, it attempts to obtain available tokens, but at this point, the remaining tokens in the token bucket are the previously configured 100 instead of the latest configured 200
’rateLimiter.tryAcquire()‘

5、That is to say, the latest restriction configuration is not effective. I checked the reason and found that it is because the Redis key that manages the remaining tokens was not deleted, resulting in the latest request still obtaining tokens from the old Redis key.

Redis key :   
     {xxx}:permits
     {xxx}:value       这个key中的值是之前配置的100，而不是最新的200。

**Comment by wynn5a** [comment]
@victoyzz 

I have attempted to reproduce the issue mentioned using the code available at https://github.com/wynn5a/redisson-rate-limiter-bug, but I was unable to reproduce the problem.

the method `org.redisson.RedissonRateLimiter#setRateAsync` specifies that `{xxx}:permits` and `{xxx}:value` will be deleted upon calling `setRate`.

Please inform me if there is anything I might have overlooked.

**Comment by handsometaoa** [comment]
你可以通过  ` RateLimiterConfig rateLimiterConfig = rateLimiter.getConfig(); `  判断限流策略是否存在更新。如果更新了，先通过 ` rateLimiter.delete(); ` 删除旧限流策略，并配置新的限流策略。

**Comment by heguanhong** [comment]
I had the same problem。

In the following code, when I request the interface once per second, the first 5 times are ok, but the sixth time, it will be blocked for more than 10 seconds

rRateLimiter = redissonClient.getRateLimiter("myTestLimiter");
rRateLimiter.trySetRate(RateType.OVERALL, 5, 1, RateIntervalUnit.SECONDS);
  @GetMapping("/myTestLimiter")
  public Result test() {
    rRateLimiter.acquire(1);
    return RespResult.success();
  }

**Comment by wynn5a** [comment]
> I had the same problem。
> 
> In the following code, when I request the interface once per second, the first 5 times are ok, but the sixth time, it will be blocked for more than 10 seconds
> 
> rRateLimiter = redissonClient.getRateLimiter("myTestLimiter"); rRateLimiter.trySetRate(RateType.OVERALL, 5, 1, RateIntervalUnit.SECONDS); @GetMapping("/myTestLimiter") public Result test() { rRateLimiter.acquire(1); return RespResult.success(); }


I am unable to replicate the issue you mentioned. For further investigation, please refer to the following GitHub repository: 
https://github.com/wynn5a/redisson-rate-limiter-bug/tree/spring

I conducted tests using a local Redis instance, it is possible that I overlooked certain details.

**Comment by heguanhong** [comment]
来信已经收到，谢谢。

**Comment by victoyzz** [comment]
> @victoyzz
> 
> I have attempted to reproduce the issue mentioned using the code available at https://github.com/wynn5a/redisson-rate-limiter-bug, but I was unable to reproduce the problem.
> 
> the method `org.redisson.RedissonRateLimiter#setRateAsync` specifies that `{xxx}:permits` and `{xxx}:value` will be deleted upon calling `setRate`.
> 
> Please inform me if there is anything I might have overlooked.

Is the version number consistent?
```
redisson-spring-boot-starter
3.22.0
```
![image](https://github.com/redisson/redisson/assets/16208144/2f09da4e-7f4d-4cad-92de-6ef58bae9630)
As shown in the figure, after retrySetRate, these keys remain in Redis and will not be actively deleted.

**Comment by wynn5a** [comment]
I have conducted further testing using Spring Boot Starter, however, I am still unable to replicate the reported issue. Kindly examine my test project on GitHub https://github.com/wynn5a/redisson-rate-limiter-bug/tree/spring-boot-starter  for a thorough evaluation.

Perhaps an unknown Redisson configuration is causing this issue. To identify the root cause, I suggest resetting the configuration to its default settings for investigation.

**Comment by mrniko** [comment]
@wynn5a 

Below is the reproducer.

```java
        RRateLimiter rateLimiter = redisson.getRateLimiter("test1");
        rateLimiter.setRate(RateType.PER_CLIENT, 5, 5, RateIntervalUnit.SECONDS);

        for (int i = 0; i < 5; i++) {
            rateLimiter.acquire();
            System.out.println("time1 " + Instant.now());
        }

        Thread.sleep(5000);

        rateLimiter.setRate(RateType.PER_CLIENT, 10, 5, RateIntervalUnit.SECONDS);

        System.out.println("---");
        Assertions.assertTimeout(Duration.ofSeconds(1), () -> {
            for (int i = 0; i < 10; i++) {
                rateLimiter.acquire();
                System.out.println("time2 " + Instant.now());
            }
        });
```

**Comment by wynn5a** [comment]
Yes, the `PRE_CLIENT` configuration seems to be the key. I will try to address this.

**Comment by mrniko** [comment]
Merged
